typescript/infra/scripts/helloworld/kathy.ts

```diff
@@ -135,17 +194,12 @@ async function main() {
     messageReceiptSeconds.labels({ origin, remote }).inc(0);
   }
   await Promise.all(
-    origins.map(async (origin) => {
-      await updateWalletBalanceMetricFor(app, origin);
+    chains.map(async (chain) => {
+      await updateWalletBalanceMetricFor(app, chain);
     }),
   );
 
-  for (
-    // in case we are restarting kathy, keep it from always running the exact same messages first
-    let currentPairingIndex = Date.now() % pairings.length;
-    ;
-    currentPairingIndex = (currentPairingIndex + 1) % pairings.length
-  ) {
+  while (true) {
```

Comment:
I am not sure I like how much logic is getting layered on this loop though. I see how it works, but it is getting pretty hard to maintain. If you have any thoughts for a way to clean it up I am curious, but otherwise it is fine for now, but if we add more logic later we should probably refactor.

Reply:
I have also been thinking about allowing concurrent sends later, so maybe we should just wait for that day to come if it does.

Reply:
Yeah honestly not sure how to make this better without a greater refactor. Concurrent sends would actually make running this in cycle-only mode much nicer too, atm sending them serially takes like > an hour which sucks. So I'd be happy to wait for a future refactor that supports concurrent sends if you're ok with that?

Reply:
sgtm